dev/dev-env/dev-shell.nix

```diff
@@ -0,0 +1,15 @@
+pkgs:
+
+pkgs.mkShell {
+  shellHook = ''
+    # A known rev from the master branch
```

Comment:
Maybe "A known rev from the master branch to test whether `nix develop` is called inside the nix-bitcoin repo". I was confused for a moment why we hardcode a rev.